operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java

```diff
@@ -156,7 +156,7 @@ void addServerIfNeeded(@Nonnull WlsServerConfig serverConfig, WlsClusterConfig c
         servers.add(serverName);
         addStartupInfo(new ServerStartupInfo(serverConfig, clusterName, server));
         addToCluster(clusterName);
-      } else if (server.isPrecreateServerService()) {
+      } else if ((!domain.isShuttingDown() || server.shouldStart(0)) && server.isPrecreateServerService()) {
```

Comment:
This condition is a bit complex. What is the intent? Can you describe it in a simple phrase that could be a method name?